airflow/jobs/scheduler_job.py

```diff
@@ -1291,7 +1291,9 @@ def _schedule_dag_run(
             self.log.error("Execution date is in future: %s", dag_run.execution_date)
             return callback
 
-        self._verify_integrity_if_dag_changed(dag_run=dag_run, session=session)
+        if not self._verify_integrity_if_dag_changed(dag_run=dag_run, session=session):
+            self.log.warning("The DAG disappeared before verifying integrity: %s. Skipping.", dag_run.dag_id)
```

Comment:
I'm wondering if we should make this a debug level logging. For example, maybe the dag disappeared during the scheduler loop but on the next loop, it appears. The user doesn't have control over it (_I think_)

Reply:
Same, this shouldn’t be a warning since the user can’t really do anything about it.

Reply:
I think this is fine to show the warning. This is a really, really rare race condition. - and realy intermittent and it can at least give  our user an indication and notice that something is wrong.

If we make it ito debug - we will never see it. And when user is asked for logs, there wil never be any suspicious thing there.

I think it should be a warning. just to let us know somthing strange is happening.